### PR TITLE
CIF-2539: Add wish list component to Venia

### PR DESCRIPTION
**Pull request by laurentiumagureanu**
## Related Issue

CIF-2539

## Motivation and Context

Enable wishlist

## How Has This Been Tested?

unit tests

## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

## Checklist:

- [x] I have signed the [Adobe Open Source CLA](http://opensource.adobe.com/cla.html).
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [x] I have read the **CONTRIBUTING** document.
- [x] I have added tests to cover my changes and the overall coverage did not decrease.
- [x] All unit tests pass on CircleCi.
- [x] I ran all tests locally and they pass.

### CODE REVIEW

File: ui.frontend/src/main/components/WishlistPage/useAddToWishlistEvent.js [hunk 1]
```diff
@@ -0,0 +1,33 @@
...
+ *    OF ANY KIND, either express or implied. See the License for the specific language
+ *    governing permissions and limitations under the License.
+ *
+ ******************************************************************************/
+import { useEventListener } from '../utils/hooks';
+import defaultOperations from '@magento/peregrine/lib/talons/Wishlist/AddToListButton/addToListButton.gql';
+import mergeOperations from '@magento/peregrine/lib/util/shallowMerge';
+import { useMutation } from '@apollo/client';
+
+const useAddToWishlistEvent = (props = {}) => {
```
Comment (Buuhuu) [inline ui.frontend/src/main/components/WishlistPage/useAddToWishlistEvent.js:19]: Should we move this to the Core Components, similar to `useAddToCartEvent` ?

Comment (laurentiumagureanu) [reply ui.frontend/src/main/components/WishlistPage/useAddToWishlistEvent.js:32]: The user can add to wishlist only if logged in.
So I call `useAddToWishlistEvent()` in the `accountTrigger` component.

Comment (Buuhuu) [reply ui.frontend/src/main/components/WishlistPage/useAddToWishlistEvent.js:32]: Calling it there is ok, but the function it self should be imported from the CIF Core Components (it is part of the glue between CIF and Peregrine, similar to`useAddToCartEvent`)

Comment (laurentiumagureanu) [reply ui.frontend/src/main/components/WishlistPage/useAddToWishlistEvent.js:32]: sure, that can work

**Review by herzog31 — APPROVED** [review]

**Review by Buuhuu — APPROVED** [review]

**Pull request merged**